### PR TITLE
1817 stock market

### PR DESCRIPTION
**Pull request by jenf**

### CODE REVIEW

File: lib/engine/share_price.rb [hunk 1]
```diff
@@ -2,51 +2,56 @@
...
                    unlimited_colors: [],
                    multiple_buy_colors: [])
       @coordinates = coordinates
       @price = price
       @color = color
-      @can_par = can_par
-      @end_game_trigger = end_game_trigger
+      @type = type
+      @can_par = type == :par
+      @end_game_trigger = type == :endgame
```
Comment (tobymao) [inline lib/engine/share_price.rb:52]: do we need vars for all of these? can we just have @type?

Comment (tobymao) [reply lib/engine/share_price.rb:52]: or perhaps a stock market space could have multiple types, not sure if that's needed now

Comment (jenf) [reply lib/engine/share_price.rb:52]: I can mod the existing ones to not need the variables.

File: lib/engine/share_price.rb [hunk 2]
```diff
@@ -2,51 +2,51 @@
...
+
+      color, type =
         case
-        when can_par
-          :red
-        when end_game_trigger
-          :blue
-        when code.include?('blk')
-          :black
+        when code.include?('p')
```
Comment (tobymao) [inline lib/engine/share_price.rb:14]: cleaner as a regex

case code
when /p/
when /c/
...

**Pull request merged**